app/src/main/java/com/example/smartshopper/MainActivity.java

```diff
@@ -42,20 +46,35 @@ protected void onCreate(Bundle savedInstanceState) {
         // to make the Navigation drawer icon always appear on the action bar
         Objects.requireNonNull(getSupportActionBar()).setDisplayHomeAsUpEnabled(true);
 
-        // Recycler View setup
-        deals = new ArrayList<>(); // initiate an empty stickers array (begins with empty recycler view)
-        // Temporary placeholder to display cards in recyclerView
-        User tempuser = new User("megaDeals100");
-        for (int i=0; i < 10; i++) {
-            String dealNumber = String.valueOf(i);
-            Deal tempDeal = new Deal("Deal"+ dealNumber , (double) (i+10), (double) (i+3),dealNumber + "description", "Target", tempuser);
-            deals.add(tempDeal);
-        }
 
-        adapter = new DealAdapter(deals, this);
+//        // Recycler View setup
+//        deals = new ArrayList<>(); // initiate an empty stickers array (begins with empty recycler view)
+//        // Temporary placeholder to display cards in recyclerView
+//        User tempuser = new User("megaDeals100");
+//        for (int i=0; i < 10; i++) {
+//            String dealNumber = String.valueOf(i);
+//            Deal tempDeal = new Deal("Deal"+ dealNumber , (double) (i+10), (double) (i+3),dealNumber + "description", "Target", tempuser);
+//            deals.add(tempDeal);
+//        }
+//
+//        adapter = new DealAdapter(deals, this);
+//        rv_dealsRecyclerView = findViewById(R.id.rv_dealsRecyclerView);
+//        rv_dealsRecyclerView.setLayoutManager(new LinearLayoutManager(this));
+//        rv_dealsRecyclerView.setAdapter(adapter);
+
+
+
         rv_dealsRecyclerView = findViewById(R.id.rv_dealsRecyclerView);
-        rv_dealsRecyclerView.setLayoutManager(new LinearLayoutManager(this));
-        rv_dealsRecyclerView.setAdapter(adapter);
+
+
+        platformHelpers.getPopularDeals(deals -> {
```

Comment:
can we put this in the platform helpers method and pass in the adapter to _getPopularDeals_ to update the adapter? -- i think its cleaner for the activity, but if u feel otherwise let me know and we can debate. lol

my refactorDO-NOT-MERGE branch for example
https://github.com/neucs5520-team-2-assignments/cs5520-team-2-group-assignments/blob/refactorCode-DO-NOT-MERGE/app/src/main/java/com/example/team2/common/PlatformMethods.java#L98-L144

Reply:
how the activity would look

https://github.com/neucs5520-team-2-assignments/cs5520-team-2-group-assignments/blob/refactorCode-DO-NOT-MERGE/app/src/main/java/com/example/team2/FirebaseRealtimeActivity.java#L46-L50

https://github.com/neucs5520-team-2-assignments/cs5520-team-2-group-assignments/blob/refactorCode-DO-NOT-MERGE/app/src/main/java/com/example/team2/FirebaseRealtimeActivity.java#L58

Reply:
okay its all internal to platform methods now